test/solc/CommandLineInterface.cpp

```diff
@@ -1412,6 +1412,130 @@ BOOST_AUTO_TEST_CASE(cli_include_paths_ambiguous_import)
 	BOOST_REQUIRE(!result.success);
 }
 
+BOOST_AUTO_TEST_CASE(cli_ethdebug_no_ethdebug_in_help)
+{
+	OptionsReaderAndMessages result = runCLI({"solc", "--help"});
+	BOOST_REQUIRE(result.stdoutContent.find("ethdebug") == std::string::npos);
+}
+
+BOOST_AUTO_TEST_CASE(cli_ethdebug_incompatible_outputs)
+{
+	TemporaryDirectory tempDir(TEST_CASE_NAME);
+	createFilesWithParentDirs({tempDir.path() / "input.sol"});
+	// output selection "asm-json" not working with debug-info "ethdebug" or "ethdebug" output selection.
+	OptionsReaderAndMessages result = runCLI({"solc", "--debug-info", "ethdebug", "--asm-json", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--ethdebug",  "--asm-json", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--debug-info", "ethdebug",  "--asm-json", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--via-ir", "--ethdebug",  "--asm-json", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--debug-info", "ethdebug",  "--asm-json", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--via-ir", "--ethdebug",  "--ir-ast-json", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--debug-info", "ethdebug",  "--ir-ast-json", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--via-ir", "--ethdebug",  "--ir-optimized-ast-json", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--debug-info", "ethdebug",  "--ir-optimized-ast-json", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--import-asm-json", tempDir.path().string() + "/input.json"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--ethdebug", "--import-asm-json", tempDir.path().string() + "/input.json"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--import-asm-json", tempDir.path().string() + "/input.json"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--via-ir", "--ethdebug",  "--asm-json", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--asm-json", tempDir.path().string() + "/input.json"}, "");
+	BOOST_REQUIRE(!result.success);
+}
+
+BOOST_AUTO_TEST_CASE(cli_ethdebug_incompatible_input_modes)
```

Comment:
This should be in `test/CommandLineParser.cpp`

Reply:
The thing is that those tests are not just checking whether the correct options were set after parsing the command lines. It is also checking for other things, e.g. error messages and actual compiler output.